docs/src/userguide/numpy_tutorial.rst

```diff
@@ -187,62 +189,95 @@ compatibility. Here's :file:`convolve_typed.pyx`. *Read the comments!*
 .. literalinclude:: ../../examples/userguide/convolve_typed.pyx
     :linenos:
 
-At this point, have a look at the generated C code for :file:`convolve1.pyx` and
-:file:`convolve2.pyx`. Click on the lines to expand them and see corresponding C.
-(Note that this code annotation is currently experimental and especially
-"trailing" cleanup code for a block may stick to the last expression in the
-block and make it look worse than it is -- use some common sense).
+.. figure:: convolve_types_html.png
 
-* .. literalinclude: convolve1.html
-* .. literalinclude: convolve2.html
+At this point, have a look at the generated C code for :file:`convolve_cy.pyx` and
+:file:`convolve_typed.pyx`. Click on the lines to expand them and see corresponding C.
 
-Especially have a look at the for loops: In :file:`convolve1.c`, these are ~20 lines
-of C code to set up while in :file:`convolve2.c` a normal C for loop is used.
+Especially have a look at the for loops: In :file:`convolve_cy.c`, these are ~20 lines
+of C code to set up while in :file:`convolve_typed.c` a normal C for loop is used.
 
 After building this and continuing my (very informal) benchmarks, I get:
 
 .. sourcecode:: ipython
 
-    In [21]: import convolve2
-    In [22]: %timeit -n2 -r3 convolve2.naive_convolve(f, g)
-    2 loops, best of 3: 828 ms per loop
+    In [19]: %timeit -n2 -r3 convolve_py.naive_convolve(f, g)
+    3.9 s ± 12.4 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
+    In [20]: %timeit -n2 -r3 convolve_cy.naive_convolve(f, g)
+    3.12 s ± 15.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
+    In [22]: %timeit -n2 -r3 convolve_typed.naive_convolve(f, g)
+    13.8 s ± 122 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
 
-Efficient indexing
-====================
+So in the end, adding types make the Cython code slower?
+
+What happend is that most of the time spend in this code is spent on line
+60. ::
+
+    value += g[smid - s, tmid - t] * f[v, w]
+
+So what made this line so much slower than in the pure Python version?
+
+``g`` and ``f`` are still NumPy arrays, so Python objects, and expect
+Python integers as indexes. Here we give C integers. So every time
+Cython reaches this line, it has to convert all the C integers to Python
+integers. Since this line is called very often, it outweight the speed
```

Comment:
I would explicitly say "to Python int objects" here, to emphasise the difference to native C values.
And "it outweighs" the speed benefits.